src/metabase/email/messages.clj

```diff
@@ -58,39 +58,45 @@
 ;; HACK: temporary workaround to postal requiring a file as the attachment
 (defn- write-byte-array-to-temp-file
   [img-bytes]
-  (let [file (java.io.File/createTempFile "metabase_pulse_image_" ".png")
-        fos (new java.io.FileOutputStream file)]
-    (.deleteOnExit file)
-    (.write fos img-bytes)
-    (.close fos)
```

Comment:
You should always use `with-open` any time there's an Object you need to manually close. `with-open` calls `.close` inside a `finally` block so it gets closed even if an exception is thrown